lasagne/layers/dnn.py

```diff
@@ -220,34 +218,25 @@ def __init__(self, incoming, num_filters, filter_size, stride=(1, 1),
         self.untie_biases = untie_biases
         self.flip_filters = flip_filters
 
-        if border_mode is not None and pad is not None:
-            raise RuntimeError("You cannot specify both 'border_mode' and "
-                               "'pad'. To avoid ambiguity, please specify "
-                               "only one of them.")
-        elif border_mode is None and pad is None:
+        if pad is None:
             # no option specified, default to valid mode
             self.pad = (0, 0)
-            self.border_mode = 'valid'
-        elif border_mode is not None:
-            if border_mode == 'valid':
+        elif isinstance(pad, basestring):
+            if pad == 'valid':
                 self.pad = (0, 0)
-                self.border_mode = 'valid'
-            elif border_mode == 'full':
+            elif pad == 'full':
                 self.pad = (self.filter_size[0] - 1, self.filter_size[1] - 1)
-                self.border_mode = 'full'
```

Comment:
You should definitely set `self.pad = 'full'` in this case, to have `dnn_conv()` handle it.